langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java

```diff
@@ -308,13 +319,26 @@ private Map<String, Object> prefillMetadata() {
     private EmbeddingMatch<TextSegment> toEmbeddingMatch(Map<String, ?> item) {
         Map<String, ?> additional = (Map<String, ?>) item.get(ADDITIONALS);
         final Metadata metadata = new Metadata();
-        if (item.get(METADATA) != null && item.get(METADATA) instanceof Map) {
-            Map<String, ?> resultingMetadata = (Map<String, ?>) item.get(METADATA);
-            for (Map.Entry<String, ?> entry : resultingMetadata.entrySet()) {
-                if (entry.getValue() != null && !NULL_VALUE.equals(entry.getValue())) {
-                    metadata.add(entry.getKey(), entry.getValue());
-                }
+        Map<String, ?> metadataMap = new HashMap();
+        if (metadataFieldName == null || metadataFieldName.isEmpty()) {
```

Comment:
is `metadataFieldName == null` condition possible?

Reply:
So no, and this is so we can keep the same functionality by defaulting to `_metadata`

I assume you want me to remove the null check here since that is the case, just felt wrong not having it with the isEmpty.